skrf/network.py

```diff
@@ -4606,51 +4973,51 @@ def innerconnect_s(A: npy.ndarray, k: int, l: int) -> npy.ndarray:
 
 ## network parameter conversion
 def s2z(s: npy.ndarray, z0: NumberLike = 50, s_def: str = S_DEF_DEFAULT) -> npy.ndarray:
-    '''
-    Convert scattering parameters [1]_ to impedance parameters [2]_
+    """
+    Convert scattering parameters [#]_ to impedance parameters [#]_
 
 
-    For power-waves, Eq.(19) from [3]:
+    For power-waves, Eq.(19) from [#Kurokawa]_:
 
     .. math::
         Z = F^{-1} (1 - S)^{-1} (S G + G^*) F
 
     where :math:`G = diag([Z_0])` and :math:`F = diag([1/2\\sqrt{|Re(Z_0)|}])`
 
-    For pseudo-waves, Eq.(74) from [4]:
+    For pseudo-waves, Eq.(74) from [#Marks]_:
 
     .. math::
         Z = (1 - U^{-1} S U)^{-1}  (1 + U^{-1} S U) G
 
     where :math:`U = \\sqrt{Re(Z_0)}/|Z_0|`
 
     Parameters
-    ------------
+    ----------
     s : complex array-like
         scattering parameters
     z0 : complex array-like or number
         port impedances.
     s_def : str -> s_def :  can be: 'power', 'pseudo' or 'traveling'
-        Scattering parameter definition : 'power' for power-waves definition [3],
-        'pseudo' for pseudo-waves definition [4].
+        Scattering parameter definition : 'power' for power-waves definition [#Kurokawa]_,
+        'pseudo' for pseudo-waves definition [#Marks]_.
         'traveling' corresponds to the initial implementation.
         Default is 'power'.
 
     Returns
-    ---------
+    -------
     z : complex array-like
         impedance parameters
 
 
 
     References
     ----------
-    .. [1] http://en.wikipedia.org/wiki/S-parameters
-    .. [2] http://en.wikipedia.org/wiki/impedance_parameters
-    .. [3] Kurokawa, Kaneyuki "Power waves and the scattering matrix", IEEE Transactions on Microwave Theory and Techniques, vol.13, iss.2, pp. 194–202, March 1965.
-    .. [4] Marks, R. B. and Williams, D. F. "A general waveguide circuit theory", Journal of Research of National Institute of Standard and Technology, vol.97, iss.5, pp. 533–562, 1992.
+    .. [#] http://en.wikipedia.org/wiki/S-parameters
+    .. [#] http://en.wikipedia.org/wiki/impedance_parameters
+    .. [#Kurokawa] Kurokawa, Kaneyuki "Power waves and the scattering matrix", IEEE Transactions on Microwave Theory and Techniques, vol.13, iss.2, pp. 194–202, March 1965.
+    .. [#Marks] Marks, R. B. and Williams, D. F. "A general waveguide circuit theory", Journal of Research of National Institute of Standard and Technology, vol.97, iss.5, pp. 533–562, 1992.
```

Comment:
Out of interest: how does this work? Will Sphinx just generate ascending label numbers for these references? And what is the difference between `[#]` with and without a label name? How does Sphinx know which one is which without a label name?

I did not find much information about this in the Sphinx docs... It would be great if you could send a link

Reply:
`[#]` are auto-numbered footnotes. So there are numbered in the order by which then appear. If you need to re-use a reference, then you should provide a label other than `#`. 

source : https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes

Reply:
Thanks a lot!